packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,20 +454,56 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
-            label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
-  );
+  const convertColumnToFilter = (column: DataTableColumn<DataTableRowData>) => {
+    if (column.enableColumnFilter === false) {
+      return;
+    }
+
+    return {
+      id: column.accessorKey as string,
+      label: column.header,
+      variant: column.filterVariant ?? "text",
+      options: column.filterSelectOptions,
+    } as DataFilter;
+  };
+
+  const isDataTableColumn = (
+    item: unknown,
+  ): item is DataTableColumn<DataTableRowData> => {
+    return (
+      !!item &&
+      typeof (item as DataTableColumn<DataTableRowData>).accessorKey !==
+        "undefined"
+    );
+  };
+
+  const isDataFilter = (item: unknown): item is DataFilter => {
+    return !!item && typeof (item as DataFilter).id !== "undefined";
+  };
+
+  const dataTableFilters = useMemo(() => {
+    const providedFilters = filtersProp || columns;
+    return providedFilters
+      .map((item) => {
+        if (typeof item === "string") {
+          const foundColumn = columns.find(
+            (column) => column.accessorKey === item,
+          );
+          return foundColumn && convertColumnToFilter(foundColumn);
+        }
+
+        if (isDataTableColumn(item)) {
+          return convertColumnToFilter(item);
+        }
+
+        if (isDataFilter(item)) {
+          return item;
+        }
+
+        return;
+      })
```

Comment:
Do we want to explicitly return `undefined` here at the end of the `.map`?
```suggestion
      .map((item) => {
        if (typeof item === "string") {
          const foundColumn = columns.find(
            (column) => column.accessorKey === item,
          );
          return foundColumn && convertColumnToFilter(foundColumn);
        }

        if (isDataTableColumn(item)) {
          return convertColumnToFilter(item);
        }

        if (isDataFilter(item)) {
          return item;
        }

        return undefined;
      })
```